From 7a264550c5f3a98b1786b1a75cff01cde1d084eb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Thu, 29 Jul 2021 17:17:43 +0200
Subject: [PATCH 5/5] shellEntry: Only mask text in password entries

When "Show Text" is locked down, we not only remove the corresponding
menu item, but also make sure the password is masked.

Except that the current code is too eager, and masks the text in
any entries.
---
 js/ui/shellEntry.js | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/js/ui/shellEntry.js b/js/ui/shellEntry.js
index c45e4545a..64b389050 100644
--- a/js/ui/shellEntry.js
+++ b/js/ui/shellEntry.js
@@ -55,11 +55,13 @@ var EntryMenu = class extends PopupMenu.PopupMenu {
                 this._passwordItem.destroy();
                 this._passwordItem = null;
             }
-            this._entry.clutter_text.set_password_char('\u25cf');
         } else if (this.isPassword && !passwordDisabled) {
             if (!this._passwordItem)
                 this._makePasswordItem();
         }
+
+        if (this.isPassword && passwordDisabled)
+            this._entry.clutter_text.set_password_char('\u25cf');
     }
 
     get isPassword() {
-- 
2.31.1

